In [37]:
import pandas as pd
import numpy as np

PREDHS = pd.read_csv("datos/Encuestas_DHS_Indicadores_Familia_Colombia.csv")
PREDHS.head()

ISO3    DataId                                          Indicator  \
0  #country+code  #meta+id                                    #indicator+name   
1            COL     78894                         Total fertility rate 15-49   
2            COL     78758  Married women currently using any method of co...   
3            COL     78759  Married women currently using any modern metho...   
4            COL     79065        Median age at first marriage [Women]: 25-49   

                  Value             Precision DHS_CountryCode    CountryName  \
0  #indicator+value+num  #indicator+precision             NaN  #country+name   
1                   3.2                     1              CO       Colombia   
2                  64.8                     1              CO       Colombia   
3                  52.4                     1              CO       Colombia   
4                  20.8                     1              CO       Colombia   

   SurveyYear    SurveyId      IndicatorId  ...  IsPreferred       SDRID  \
0  #date+year  #survey+id  #indicator+code  ...          NaN         NaN   
1        1986   CO1986DHS    FE_FRTR_W_TFR  ...          1.0  FEFRTRWTFR   
2        1986   CO1986DHS    FP_CUSM_W_ANY  ...          1.0  FPCUSMWANY   
3        1986   CO1986DHS    FP_CUSM_W_MOD  ...          1.0  FPCUSMWMOD   
4        1986   CO1986DHS    MA_AAFM_W_M2B  ...          1.0  MAAAFMWM2B   

   RegionId  SurveyYearLabel SurveyType DenominatorWeighted  \
0       NaN              NaN        NaN                 NaN   
1       NaN           1986.0        DHS                 NaN   
2       NaN           1986.0        DHS              2850.0   
3       NaN           1986.0        DHS              2850.0   
4       NaN           1986.0        DHS                 NaN   

  DenominatorUnweighted CILow  CIHigh  LevelRank  
0                   NaN   NaN     NaN        NaN  
1                   NaN   NaN     NaN        NaN  
2                2848.0   NaN     NaN        NaN  
3                2848.0   NaN     NaN        NaN  
4                   NaN   NaN     NaN        NaN  

[5 rows x 29 columns]

# Exploración Inicial de los Datos 🔎
Para analizar la información de mi dataset, su tamaño y las columnas. 

In [38]:
PREDHS.info()
PREDHS.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 138 entries, 0 to 137
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   ISO3                    138 non-null    object 
 1   DataId                  138 non-null    object 
 2   Indicator               138 non-null    object 
 3   Value                   138 non-null    object 
 4   Precision               138 non-null    object 
 5   DHS_CountryCode         137 non-null    object 
 6   CountryName             138 non-null    object 
 7   SurveyYear              138 non-null    object 
 8   SurveyId                138 non-null    object 
 9   IndicatorId             138 non-null    object 
 10  IndicatorOrder          137 non-null    float64
 11  IndicatorType           137 non-null    object 
 12  CharacteristicId        137 non-null    float64
 13  CharacteristicOrder     137 non-null    float64
 14  CharacteristicCategory  137 non-null    ob

(138, 29)

In [39]:
PREDHS.head()

ISO3    DataId                                          Indicator  \
0  #country+code  #meta+id                                    #indicator+name   
1            COL     78894                         Total fertility rate 15-49   
2            COL     78758  Married women currently using any method of co...   
3            COL     78759  Married women currently using any modern metho...   
4            COL     79065        Median age at first marriage [Women]: 25-49   

                  Value             Precision DHS_CountryCode    CountryName  \
0  #indicator+value+num  #indicator+precision             NaN  #country+name   
1                   3.2                     1              CO       Colombia   
2                  64.8                     1              CO       Colombia   
3                  52.4                     1              CO       Colombia   
4                  20.8                     1              CO       Colombia   

   SurveyYear    SurveyId      IndicatorId  ...  IsPreferred       SDRID  \
0  #date+year  #survey+id  #indicator+code  ...          NaN         NaN   
1        1986   CO1986DHS    FE_FRTR_W_TFR  ...          1.0  FEFRTRWTFR   
2        1986   CO1986DHS    FP_CUSM_W_ANY  ...          1.0  FPCUSMWANY   
3        1986   CO1986DHS    FP_CUSM_W_MOD  ...          1.0  FPCUSMWMOD   
4        1986   CO1986DHS    MA_AAFM_W_M2B  ...          1.0  MAAAFMWM2B   

   RegionId  SurveyYearLabel SurveyType DenominatorWeighted  \
0       NaN              NaN        NaN                 NaN   
1       NaN           1986.0        DHS                 NaN   
2       NaN           1986.0        DHS              2850.0   
3       NaN           1986.0        DHS              2850.0   
4       NaN           1986.0        DHS                 NaN   

  DenominatorUnweighted CILow  CIHigh  LevelRank  
0                   NaN   NaN     NaN        NaN  
1                   NaN   NaN     NaN        NaN  
2                2848.0   NaN     NaN        NaN  
3                2848.0   NaN     NaN        NaN  
4                   NaN   NaN     NaN        NaN  

[5 rows x 29 columns]

### Buscando valores nulos

In [40]:
PREDHS.isnull().sum()

ISO3                        0
DataId                      0
Indicator                   0
Value                       0
Precision                   0
DHS_CountryCode             1
CountryName                 0
SurveyYear                  0
SurveyId                    0
IndicatorId                 0
IndicatorOrder              1
IndicatorType               1
CharacteristicId            1
CharacteristicOrder         1
CharacteristicCategory      1
CharacteristicLabel         1
ByVariableId                0
ByVariableLabel            78
IsTotal                     1
IsPreferred                 1
SDRID                       1
RegionId                  138
SurveyYearLabel             1
SurveyType                  1
DenominatorWeighted        56
DenominatorUnweighted      56
CILow                     109
CIHigh                    109
LevelRank                 138
dtype: int64

Para conocer el porcentaje de valores nulos por columna. 

In [41]:
nvalorespct = PREDHS.isnull().mean() * 100
nvalorespct

ISO3                        0.000000
DataId                      0.000000
Indicator                   0.000000
Value                       0.000000
Precision                   0.000000
DHS_CountryCode             0.724638
CountryName                 0.000000
SurveyYear                  0.000000
SurveyId                    0.000000
IndicatorId                 0.000000
IndicatorOrder              0.724638
IndicatorType               0.724638
CharacteristicId            0.724638
CharacteristicOrder         0.724638
CharacteristicCategory      0.724638
CharacteristicLabel         0.724638
ByVariableId                0.000000
ByVariableLabel            56.521739
IsTotal                     0.724638
IsPreferred                 0.724638
SDRID                       0.724638
RegionId                  100.000000
SurveyYearLabel             0.724638
SurveyType                  0.724638
DenominatorWeighted        40.579710
DenominatorUnweighted      40.579710
CILow                      78.985507
C

## Limpieza de Datos 🧹
**Problemas Identificados:**
1. Columnas con demasiados valores nulos:
- **RegionId y LevelRank:** tienen 100% de valores nulos, los eliminaré ya que no aportan ninguna información relevante.
- **ByVariableLabel**: tiene 56.52% de valores nulos. 
- **DenominatorWeighted y DenominatorUnweighted:** tienen 40.58% de valores nulos.
-  **CILow y CIHigh:** tienen 78.99% de valores nulos.
- **DHS_CountryCode, IndicatorOrder, IndicatorType, CharacteristicId, CharacteristicOrder:** tienen valores nulos en menos del 1%, los voy a inputar. 